File: 83. Remove Duplicates from Sorted List/deleteDuplicates.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null){
            return head;
        }
        else{
            ListNode help = head;
            while (help.next != null){
                if (help.val == help.next.val){
                    help.next = help.next.next;
                }
                else{
                    help = help.next;
                }
            }
            return head;
        }
    }
}

File: 83. Remove Duplicates from Sorted List/deleteDuplicates.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        ptr = head
        while ptr and ptr.next:
            while ptr.next and ptr.val == ptr.next.val:
                ptr.next = ptr.next.next
            ptr = ptr.next
        return head
        